ocaml/testsuite/tests/typing-higher-jkinds/top.ml

```diff
@@ -0,0 +1,66 @@
+(* TEST
+ flags = "-extension layouts_alpha";
+ expect;
+*)
+
+type t : top
+[%%expect {|
+type t : top
+|}]
+
+type t : top => value
+[%%expect {|
+type t : top => value
+|}]
+
+module M : sig
+  type t : top
+end = struct
+  type t : top
+end
+[%%expect {|
+module M : sig type t : top end
+|}]
+
+module M : sig
+  type t : top
+end = struct
+  type t : value
+end
+[%%expect {|
+module M : sig type t : top end
+|}]
+
+module M : sig
+  type t : value => value
+end = struct
+  type t : value => top
+end
+[%%expect {|
+Lines 3-5, characters 6-3:
+3 | ......struct
+4 |   type t : value => top
+5 | end
+Error: Signature mismatch:
+       Modules do not match:
+         sig type t : value => top end
+       is not included in
+         sig type t : value => value end
+       Type declarations do not match:
+         type t : value => top
+       is not included in
+         type t : value => value
+       The kind of the first is ((value) => top)
```

Comment:
I think this is from an earlier PR, but this test shows we're printing the argument kinds with too many parentheses sometimes.